src/messages/tx.js

```diff
@@ -2,18 +2,89 @@ var struct = require('varstruct')
 var varint = require('varuint-bitcoin')
 var types = require('../types')
 
-// TODO: add segwit
-module.exports = struct([
-  { name: 'version', type: struct.Int32LE },
-  { name: 'ins', type: struct.VarArray(varint, struct([
-    { name: 'hash', type: types.Buffer32 },
-    { name: 'index', type: struct.UInt32LE },
-    { name: 'script', type: types.VarBuffer },
-    { name: 'sequence', type: struct.UInt32LE }
-  ])) },
-  { name: 'outs', type: struct.VarArray(varint, struct([
-    { name: 'valueBuffer', type: types.Buffer8 },
-    { name: 'script', type: types.VarBuffer }
-  ])) },
-  { name: 'locktime', type: struct.UInt32LE }
-])
+var inputs = struct.VarArray(varint, struct([
+  { name: 'hash', type: types.Buffer32 },
+  { name: 'index', type: struct.UInt32LE },
+  { name: 'script', type: types.VarBuffer },
+  { name: 'sequence', type: struct.UInt32LE }
+]))
+
+var outputs = struct.VarArray(varint, struct([
+  { name: 'valueBuffer', type: types.Buffer8 },
+  { name: 'script', type: types.VarBuffer }
+]))
+
+function encode (value, buffer, offset) {
+  if (!buffer) buffer = new Buffer(encodingLength(value))
+  if (!offset) offset = 0
+  encode.bytes = 0
+  // version
+  struct.Int32LE.encode(value.version, buffer, offset + encode.bytes)
+  encode.bytes += struct.Int32LE.encode.bytes
+  // marker and flags if segwit
+  if (value.segwit) {
```

Comment:
Why require a `segwit` property, couldn't we imply it is segwit just by having the `flags` and marker?

It might be OK to keep it in the output of `decode` though.